quad_pyblish_module/plugins/photoshop/publish/validate_blendmode.py

```diff
@@ -0,0 +1,69 @@
+import pyblish.api
+from openpype.pipeline.publish import (
+    ValidateContentsOrder,
+    PublishXmlValidationError,
+    OptionalPyblishPluginMixin
+)
+from openpype.hosts.photoshop import api as photoshop
+
+
+class ValidateBlendModeRepair(pyblish.api.Action):
+    """Repair the instance asset."""
+
+    label = "Repair"
+    icon = "wrench"
+    on = "failed"
+
+    def process(self, context, plugin):
+
+        stub = photoshop.stub()
+        failed = context.data['transientData'][ValidateBlendMode.__name__]
+
+        for layer, info in failed.items():
+            stub.set_blendmode(layer_name=layer, blendMode_name=info["defaultBlendMode"])
+
+
+class ValidateBlendMode(OptionalPyblishPluginMixin,
+    pyblish.api.ContextPlugin):
+    """Validate if the blendMode is set properly on Layers, NORMAL, and Groups, PASSTHROUGH
+    """
+
+    label = "Validate BlendMode"
+    hosts = ["photoshop"]
+    optional = True
+    actions = [ValidateBlendModeRepair]
+    order = ValidateContentsOrder
+
+    def process(self, context):
+
+        if not self.is_active(context.data):
+            return
+        
+        PASSTHROUGH = "passThrough"
+        NORMAL = "normal"
+        returnDict =  {}
+        msg = ""
+
+        stub = photoshop.stub()
+        layers = stub.get_layers()
+        
+        for layer in layers:
+            layerDict = {}
+            if (layer.group and layer.blendMode != PASSTHROUGH) or (not layer.group and layer.blendMode != NORMAL):
```

Comment:
Et ici faire `if (layer.group and layer.blendMode != BlendMode.PassThrough) or (not layer.group and layer.blendMode != BlendMode.NormalBlend):`

(Voir https://github.com/loonghao/photoshop-python-api/blob/main/photoshop/api/enumerations.py#L91 )

Reply:
il y a une logique `typeIDToStringID(desc.getEnumerationValue(stringIDToTypeID('mode')));` pour obtenir une string, voir si il y a la même pour avoir un id (int)

Reply:
Comme vu dans la pr sur [OpenPype](https://github.com/quadproduction/OpenPype/pull/699), il n'y a pas de lien entre les classes et fonction dans le jsx et celles du py, les ID des blendModes ne correspondent pas entre l'api jsx et l'api python de Photoshop.
En passant par un eval('BlendMode.' + blendModeName.toUpperCase()), on va conserver la transmission d'info par str je pense

Reply:
Yes